core/src/test/java/org/apache/iceberg/rest/TestHTTPClient.java

```diff
@@ -265,4 +281,17 @@ public boolean equals(Object o) {
       return Objects.equals(id, item.id) && Objects.equals(data, item.data);
     }
   }
+
+  public static class TestHttpRequestInterceptor implements HttpRequestInterceptor {
```

Comment:
I'd prefer not to expose this as a feature.

Reply:
I'm leaving this in so that we can test the loading logic, but it's no longer possible to provide an alternate implementation.

Reply:
Sounds good.